pilot/pkg/bootstrap/servicecontroller.go

```diff
@@ -81,6 +81,11 @@ func (s *Server) initKubeRegistry(args *PilotArgs) (err error) {
 		caBundlePath = args.ServerOptions.TLSOptions.CaCertFile
 	}
 
+	mcsSettings := kubecontroller.MCSSettings{
+		EnableServiceExport: false, //todo change to s.environment.Mesh().EnableMcsServiceExport once https://github.com/istio/api/pull/1785 is merged,
+		PushContext:         s.environment.PushContext,
```

Comment:
I think what you want is the environment, not the pushcontext, actually. The pushcontext is set as a property of the environment, but it can be overwritten and is therefore not stable for concurrent access. Environment should be stable throughout the life of the process.